scripts/generate_icon_types.js

```diff
@@ -0,0 +1,234 @@
+// Script to generate TypeScript icon type definitions from filesystem
+import { existsSync, readdirSync, statSync, writeFileSync, mkdirSync } from 'fs';
+import { resolve, join, dirname } from 'path';
+import { fileURLToPath } from 'url';
+
+// ESM equivalent of __dirname
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+// Paths
+const iconsOutlinePath = resolve(__dirname, '../assets/icons/outline');
+const iconsSolidPath = resolve(__dirname, '../assets/icons/solid');
+const buildPath = resolve(__dirname, '../build');
+const outputJsPath = join(buildPath, 'icons.js');
+const outputDtsPath = join(buildPath, 'icons.d.ts');
+
+/**
+ * Recursively get all SVG files from a directory
+ * @param {string} dirPath - Directory to scan
+ * @returns {string[]} - Array of icon names (without .svg extension)
+ */
+function getIconNames(dirPath) {
+  const iconNames = [];
+
+  function traverse(currentPath) {
+    if (!existsSync(currentPath)) {
+      return;
+    }
+
+    const items = readdirSync(currentPath);
+
+    items.forEach(item => {
+      const fullPath = join(currentPath, item);
+      const stat = statSync(fullPath);
+
+      if (stat.isDirectory()) {
+        // Recursively traverse subdirectories
+        traverse(fullPath);
+      } else if (item.endsWith('.svg')) {
+        // Extract filename without extension
+        const iconName = item.replace('.svg', '');
+        iconNames.push(iconName);
+      }
```

Comment:
e simplificamos esse trecho, tirando o `statSync` para cada item, resultando em menos I/O de disco

```suggestion
      if (item.isDirectory()) {
        // Recursively traverse subdirectories
        traverse(fullPath);
      } else if (item.isFile() && item.name.endsWith('.svg')) {
        // Extract filename without extension
       const iconName = item.name.replace('.svg', '');
        iconNames.push(iconName);
      }
```